### ISSUE TITLE
[torchax] make_train_loop to support torch optimizers

### ISSUE THREAD

**Issue by qihqi** [pytorch/xla#9301]
## 🚀 Feature
Currently the [`make_train_loop`] (https://github.com/pytorch/xla/blob/master/torchax/torchax/train.py#L14) function takes in an `optax` Optimizer for doing the trianing loop. It is desirable to also take in torch optimizer.

The difference between those 2 types of optimizer is that torch optimizer acts on the `.grad` field of each paramters, where optax optimizer 
acts on a pytree of gradients.

To make torch optimizer work, roughly
around the lines https://github.com/pytorch/xla/blob/master/torchax/torchax/train.py#L47

We want to insert the gradient computed by `grad_fn` above to `tensor.grad` of each weight:
along the lines of 

```
for tensor, grad in zip(pytree.flatten(weights), pytree.flatten(gradient)):
  tensor.grad = grad

optimizer.step()
```

The above accomplishes the update.

Now, because the step function returned needs to take a pytree of optimizer state and return optimizer state,
so need to add loading and unloading of optimizer state inside of that.

I.e. 

```
optimizer.load_state_dict(opt_state)
for tensor, grad in zip(pytree.flatten(weights), pytree.flatten(gradient)):
  tensor.grad = grad

optimizer.step()
...

return loss, weights, opt_state
``` 

After that, one would able to use both optax optimizer as well as torch optimizers, probably also add options / configs to this function
